File: 04 - Java/HelloWorld.java
// Día 4: Java
// Clase en vídeo: https://youtu.be/W86KTBSiX2o

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

// Esta clase se llama HelloWorld en vez de Main para mantener la coherencia con el proyecto
public class HelloWorld {
    public static void main(String[] args) {

        // Hola mundo

        System.out.println("Hola, Java!");

        // Variables

        String myString = "Esto es una cadena de texto";
        myString = "Aquí cambio el valor de la cadena de texto";
        System.out.println(myString);

        Integer myInt = 7;
        myInt = myInt + 4;
        System.out.println(myInt);
        System.out.println(myInt - 1);

        Double myDouble = 6.5;
        System.out.println(myDouble);

        Float myFloat = 6.5f;
        System.out.println(myFloat);

        System.out.println(myDouble + " " + myString + " " + myFloat + " " + myInt);

        Boolean myBoolean = true;
        myBoolean = false;
        System.out.println(myBoolean);

        myBoolean = null;
        System.out.println(myBoolean);

        // Constantes

        final String myConstant = "Esto es una cadena de texto constante";

        // Control de flujo

        myFloat = null;
        if (myFloat != null) {
            System.out.println(myFloat + 10);
            System.out.println("My Float es distinto de nulo");
        } else {
            System.out.println("My Float es nulo");
        }

        if (myInt == 11) {
            System.out.println("myInt == 11");
        } else if (myInt == 12) {
            System.out.println("myInt == 12");
        } else {
            System.out.println("El valor no es ni 11 ni 12");
        }

        // Listas

        List<String> myList = new ArrayList();
        myList.add(myString);
        myList.add(myInt.toString());
        System.out.println(myList);

        // Mapas

        Map<String, String> myMap = new HashMap();
        myMap.put("string", myString);
        myMap.put("int", myInt.toString());
        System.out.println(myMap);
        System.out.println(myMap.get("int"));

        // Bucles

        for (int index = 0; index < myList.size(); index++) {
            System.out.println(myList.get(index));
        }

        // Clases

        HelloWorld myMain = new HelloWorld();
        System.out.println(myMain.myFunction(5, 2));
        System.out.println(myMain.myFunction(3, 126389));
    }

    // Funciones

    public int myFunction(int myFirstNumber, int mySecondNumber) {
        return myFirstNumber + mySecondNumber;
    }

}

File: 10 - PHP/hello_world.php
# Día 10: PHP
# Clase en vídeo: https://youtu.be/nPCJAx5c1uE

<?php

// Hola mundo
echo "Hola, PHP \n";

/*
Este
es
un
comentario
*/

// Variables

$my_string = "Esto es una cadena de texto";
$my_string = "Aquí cambio el valor de la cadena de texto";
echo $my_string . "\n";
echo gettype($my_string) . "\n";

$my_string = 6; // Tipado dinámico
echo $my_string . "\n";
echo gettype($my_string) . "\n";

$my_string = "Esto es una cadena de texto";

$my_int = 7;
$my_int = $my_int + 4;
echo $my_int . "\n";
echo $my_int - 1 . "\n";
echo $my_int . "\n";
echo gettype($my_int) . "\n";

$my_double = 6.5;
echo gettype($my_double) . "\n";
echo $my_int + $my_double . "\n";
// echo $my_int + $my_double + $my_string . "\n";

$my_bool = false;
echo $my_bool . "\n";
$my_bool = true;
echo $my_bool . "\n";
echo gettype($my_bool) . "\n";

echo "El valor de mi integer es $my_int y el de mi boolen es $my_bool.\n";

// Constantes

const MY_CONSTANT = "Valor de la constante";
echo MY_CONSTANT . "\n";

// Listas

$my_array = [$my_string, $my_int, $my_double];
echo gettype($my_array) . "\n";
echo $my_array[0] . "\n";
array_push($my_array, $my_bool);
print_r($my_array);
// echo $my_array[4] . "\n"; // Error

// Diccionarios

$my_dict = array("string" => $my_string, "int" => $my_int, "bool" => $my_bool);
echo gettype($my_dict) . "\n";
print_r($my_dict);
echo $my_dict["int"] . "\n";

// Sets

array_push($my_array, "Brais");
array_push($my_array, "Brais");
print_r($my_array);
print_r(array_unique($my_array));

// Control de flujo

for ($index = 0; $index <= 10; $index++) {
    echo $index . "\n";
}

foreach ($my_array as $my_item) {
    echo $my_item . "\n";
}

$index = 0;
while ($index <= sizeof($my_array) - 1) {
    echo $my_array[$index] . "\n";
    $index++;
}

if ($my_int == 11 && $my_string == "Hola") {
    echo "El valor es 11\n";
} elseif ($my_int == 12 || $my_string == "Hola") {
    echo "El valor es 12\n";
} else {
    echo "El valor no es 11\n";
}

// Funciones

function print_number(int $my_number) {
    echo $my_number . "\n";
}

print_number(10.5);
print_number(11);
print_number(12);

// Clases

class MyClass {
    public $name;
    public $age;

    function __construct($name, $age) {
        $this->name = $name;
        $this->age = $age;
    }
}

$my_class = new MyClass("Brais", 36);
print_r($my_class);
echo $my_class->name . "\n";
$my_class->name = "MoureDev";
echo $my_class->name . "\n";
echo gettype($my_class) . "\n";